scripts/compile.js

```diff
@@ -0,0 +1,229 @@
+require('dotenv').config()
+const { exec } = require('child_process')
+const { promisify } = require('util')
+const fse = require('fs-extra')
+var fs = require('fs');
+const { join } = require('path');
+var dir = './tmp';
+
+const existsAsync = promisify(fs.exists)
+const mkdirAsync = promisify(fs.mkdir)
+const unlinkAsync = promisify(fs.unlink)
+const execAsync = promisify(exec)
+
+
+const command = ({ contract, source, include, dir }) => {
+  const volume = dir
+  let cmd = ""
+  let inc = include == "" ? "./include" : include
+  
+  if (process.env.COMPILER === 'local') {
+    cmd = "eosio-cpp -abigen -I "+ inc +" -contract " + contract + " -o ./artifacts/"+contract+".wasm "+source;
+  } else {
+    cmd = `docker run --rm --volume ${volume}:${volume} -w ${volume} eosio/key-value-example:v1.0.0 /bin/bash -c "cd build && cmake .. && make"`
```

Comment:
I'm using other image for building - I don't mind using either one, but we should probably stick to one: https://github.com/hypha-dao/voice-token/blob/master/.github/workflows/build.yml#L9

Also, we could probably "glue" this part with cmake, There could be a way (not sure) to tell cmake to build and execute this test. It would be cleaner IMO

Reply:
I don't mind changing this to anything else

@josejulio uses docker://josejulio/hypha-builder:release_1.7.x

Happy to use it here too

Reply:
@IanMendozaJaimes I am using local, so will add this in the local command too

@josejulio I had some issues with docker images in the past but that is 2 years ago - chances are docker is OK now on mac ;)

Reply:
Local install is much more performant though

Reply:
It is! is just that the tools are only supported on ubuntu and something else (can't recall right now). 
Maybe this could check if eosio-cpp is found in path (using `which` or something else) and execute it and fallback to the docker.